script.js

```diff
@@ -1,4 +1,216 @@
+// Project 1: Scissors Paper Stone
+
+// Define constants to use in multiple functions to avoid bugs due to typos.
+var SCISSORS = "scissors";
+var PAPER = "paper";
+var STONE = "stone";
+var numRolls = 0;
+var numWins = 0;
+var winPercent = "";
+var winRecord = "";
+var userName = "";
+var gameMode = "waiting for username";
+var mostRecentWinner = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == "waiting for username") {
+    userName = input;
+    gameMode = "normal";
+    return `Hello ${userName}! You can start playing by entering "scissors", "paper" or "stone". `;
+  } else {
+    if (gameMode == "normal") {
+      return spsGame(userName, input);
+    }
+    if (gameMode == "reverse") {
+      return reversedSPSGame(userName, input);
+    }
+    if (gameMode == "korean") {
+      return koreanSPSGame(userName, input);
+    }
+  }
+};
+
+// normal SPS game
+var spsGame = function (userName, input) {
+  if (input == "reverse") {
+    gameMode = "reverse";
+    return `The rules of the SPS game have been reversed`;
+  } else if (input == "korean") {
+    gameMode = "korean";
+    return `The Korean SPS game will start now.`;
+  }
+
+  var comOutput = generateComOutput();
+  var standardMessage = `${userName}, you chose ${input} <br> The computer chose ${comOutput} <br>`;
+  var endMessage = `Now you can type "scissors", "paper" or "stone" to play another round!`;
+  numRolls = numRolls + 1;
+
+  //Computer and User draw when they choose the same object
+  if (input == comOutput) {
+    winRecord = `${numWins} out of ${numRolls}`;
+    var ProgressMessage = generateProgressMessage(numWins, numRolls);
+
+    return `${standardMessage} <br> You draw! <br> <br> So far ${userName}, you've been winning ${winRecord} turns. <br> <br> ${endMessage} ${ProgressMessage}`;
+  }
+
+  //scissors beats paper, paper beats stone, and stone beats scissors.
+
+  //Computer wins and user loses
+  if (
+    (comOutput == SCISSORS && input == PAPER) ||
+    (comOutput == PAPER && input == STONE) ||
+    (comOutput == STONE && input == SCISSORS)
+  ) {
```

Comment:
It seems that this logic is repeated several times. This is a tell tale sign that you could consider making it a function. An example would be checkWinner(input, comOutput)